x/poolmanager/router.go

```diff
@@ -548,6 +548,43 @@ func (k Keeper) AllPools(
 	return sortedPools, nil
 }
 
+// ListPoolsByDenom returns all pools by denom sorted by their ids
+// from every pool module registered in the
+// pool manager keeper.
+func (k Keeper) ListPoolsByDenom(
+	ctx sdk.Context,
+	Denom string,
+) ([]types.PoolI, error) {
+	less := func(i, j types.PoolI) bool {
+		return i.GetId() < j.GetId()
+	}
+	//	Allocate the slice with the exact capacity to avoid reallocations.
+	poolCount := k.GetNextPoolId(ctx)
+	sortedPools := make([]types.PoolI, 0, poolCount)
+	for _, poolModule := range k.poolModules {
+		currentModulePools, err := poolModule.GetPools(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		var poolsByDenom []types.PoolI
+		for _, pool := range currentModulePools {
+			coins, err := k.GetTotalPoolLiquidity(ctx, pool.GetId())
+			if err != nil {
+				return nil, err
+			}
+			if coins.AmountOf(Denom).GT(osmomath.ZeroInt()) {
+				if err != nil {
+					return nil, err
+				}
```

Comment:
why does it error if some pool does not have the denom we want? doesn't it simply need to `continue` in this case?